mlir/lib/Dialect/SparseTensor/IR/SparseTensorDialect.cpp

```diff
@@ -818,6 +824,53 @@ AffineMap mlir::sparse_tensor::inverseBlockSparsity(AffineMap dimToLvl,
   return dimToLvl.get(dimToLvl.getNumResults(), 0, lvlExprs, context);
 }
 
+SmallVector<unsigned> mlir::sparse_tensor::getBlockSize(AffineMap dimToLvl) {
+  assert(isBlockSparsity(dimToLvl) &&
+         "expected dimToLvl to be block sparsity for calling getBlockSize");
+  SmallVector<unsigned> blockSize;
+  for (auto result : dimToLvl.getResults()) {
+    if (auto binOp = result.dyn_cast<AffineBinaryOpExpr>()) {
+      if (result.getKind() == AffineExprKind::Mod) {
+        blockSize.push_back(
+            binOp.getRHS().dyn_cast<AffineConstantExpr>().getValue());
+      }
+    } else {
+      blockSize.push_back(1);
```

Comment:
note that 1 could still be used for an explicit floor/mod blocking by 1; this is weird, but would expand that dimension into two. In contrast, a non-expanding dimension does not change rank. So arguable we should return 0...

but it is detail we can debate later too

Reply:
Yeah, that's true. But how should we handle cases for 2:4 like below? It's technically 1x4 so block size for `i` should be 1?
#NV_24 = #sparse_tensor.encoding<{
  map = ( i, j ) ->
  ( i                  : dense,
    j floordiv 4 : dense,
    j mod 4      : block2_4
  )
}>